src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs

```diff
@@ -247,12 +249,21 @@ protected static bool IsInvalidArgument(IReadOnlyList<object> list)
             return payloads;
         }
 
-        private async Task WriteCoreAsync<T>(T message, Func<T, Task> task) where T : ServiceMessage, IMessageWithTracingId
+        private async Task WriteCoreAsync<T>(T message, Func<T, Task> task, string connectionId = null) where T : ServiceMessage, IMessageWithTracingId
         {
             try
             {
                 await task(message);
             }
+            catch (TimeoutException ex)
+            {
+                // Regard the case connection already disconnected when send group message got timeout as success
+                if (connectionId == null || _clientConnectionManager.ClientConnections.TryGetValue(connectionId, out var connection))
```

Comment:
is `connectionId == null` an `ArgumentNullException`? can we actually get a null connection id?

Reply:
Common send message doesn't have connectionId information and it's default set null. Now only join/leave group is using ackable message.

Reply:
How about checking:
```
if (message is Join/LeaveGroupWithAckMessage msg && _clientConnectionManager.ClientConnections.TryGetValue(msg.ConnectionId, out var connection))
{
....
}
```
then no need to change the function parameters